File: Server/route/userCTFChallenge.go
package route

import (
	"github.com/CoolCoolTomato/MatrilxArena/Server/api"
	"github.com/CoolCoolTomato/MatrilxArena/Server/middleware"
	"github.com/gin-gonic/gin"
)

func SetUserCTFChallengeRoute(r *gin.Engine) {
	userChallenge := r.Group("/userCTFChallenge")
	userChallenge.Use(middleware.JWTAuthMiddleware())
	{
		userChallenge.POST("/GetUserCTFChallengeList", api.GetUserCTFChallengeList)
		userChallenge.POST("/ResetUserCTFChallenge", api.ResetUserCTFChallenge)
		userChallenge.POST("/CheckCTFChallengeFlag", api.CheckCTFChallengeFlag)
	}
}

File: Web/src/api/userCTFChallenge.js
import {apiClient, handleResponse, handleError} from "./api.js"

function GetUserCTFChallengeList(data) {
  return apiClient.post("/userCTFChallenge/GetUserCTFChallengeList", data).then(handleResponse).catch(handleError)
}

function ResetUserCTFChallenge(data) {
  return apiClient.post("/userCTFChallenge/ResetUserCTFChallenge", data).then(handleResponse).catch(handleError)
}

function CheckCTFChallengeFlag(data) {
  return apiClient.post("/userCTFChallenge/CheckCTFChallengeFlag", data).then(handleResponse).catch(handleError)
}

const userCTFChallengeApi = {
  GetUserCTFChallengeList,
  ResetUserCTFChallenge,
  CheckCTFChallengeFlag
}

export default userCTFChallengeApi
